Error on logout

**Describe the bug**
When clicking logout button from a TI profile, I am seeing an error message. 

**To Reproduce**
Steps to reproduce the behavior:

1. Login as a TI
2. Click on "Logout"
3. see the error

**Expected behavior**
Logout should be smooth without errors

**Screenshots**
Error screen
![Image](https://github.com/user-attachments/assets/39e9f6e1-e510-4170-9315-0c892e0045f8)

Error Message
java.lang.NullPointerException: Cannot invoke "auth.oidc.IdTokens.getIdToken(String)" because "idTokens" is null
	at auth.oidc.CiviformOidcLogoutActionBuilder.getIdTokenForAccount(CiviformOidcLogoutActionBuilder.java:108)
	at auth.oidc.CiviformOidcLogoutActionBuilder.getLogoutAction(CiviformOidcLogoutActionBuilder.java:141)
	at org.pac4j.core.client.IndirectClient.getLogoutAction(IndirectClient.java:168)
	at org.pac4j.core.engine.DefaultLogoutLogic.perform(DefaultLogoutLogic.java:121)
	at modules.CiviFormLogoutLogic.perform(CiviFormLogoutLogic.java:37)
	at org.pac4j.play.LogoutController.lambda$logout$0(LogoutController.java:55)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(Unknown Source)
	at play.core.j.ClassLoaderExecutionContext.$anonfun$execute$1(ClassLoaderExecutionContext.scala:64)
	at org.apache.pekko.dispatch.TaskInvocation.run(AbstractDispatcher.scala:59)
	at org.apache.pekko.dispatch.ForkJoinExecutorConfigurator$PekkoForkJoinTask.exec(ForkJoinExecutorConfigurator.scala:57)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(Unknown Source)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(Unknown Source)
	at java.base/java.util.concurrent.ForkJoinPool.scan(Unknown Source)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(Unknown Source)
Caused by: java.lang.NullPointerException: Cannot invoke "auth.oidc.IdTokens.getIdToken(String)" because "idTokens" is null
	at auth.oidc.CiviformOidcLogoutActionBuilder.getIdTokenForAccount(CiviformOidcLogoutActionBuilder.java:108)
	at auth.oidc.CiviformOidcLogoutActionBuilder.getLogoutAction(CiviformOidcLogoutActionBuilder.java:141)
	at org.pac4j.core.client.IndirectClient.getLogoutAction(IndirectClient.java:168)
	at org.pac4j.core.engine.DefaultLogoutLogic.perform(DefaultLogoutLogic.java:121)
	at modules.CiviFormLogoutLogic.perform(CiviFormLogoutLogic.java:37)
	at org.pac4j.play.LogoutController.lambda$logout$0(LogoutController.java:55)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(Unknown Source)
	at play.core.j.ClassLoaderExecutionContext.$anonfun$execute$1(ClassLoaderExecutionContext.scala:64)
	... 7 common frames omitted

--- a/server/app/auth/oidc/CiviformOidcLogoutActionBuilder.java
+++ b/server/app/auth/oidc/CiviformOidcLogoutActionBuilder.java
@@ -102,6 +102,9 @@ public final class CiviformOidcLogoutActionBuilder extends OidcLogoutActionBuild
     }
 
     IdTokens idTokens = account.get().getIdTokens();
+    if (idTokens == null) {
+      return Optional.empty();
+    }
 
     // When we build the logout action, we do not remove the id token. We leave it in place in case
     // of transient logout failures. Expired tokens are purged at login time instead.

--- a/server/test/auth/oidc/CiviformOidcLogoutActionBuilderTest.java
+++ b/server/test/auth/oidc/CiviformOidcLogoutActionBuilderTest.java
@@ -199,4 +199,124 @@ public class CiviformOidcLogoutActionBuilderTest extends ResetPostgres {
     assertThat(queryParamValue(locationUri, "custom_target_url_parameter_name"))
         .hasValue(targetUrl);
   }
+
+  @Test
+  public void testBuilderLogsOutIfIdTokensIsNull() throws URISyntaxException {
+    // Enable enhanced logout for admins.
+    Config civiformConfig =
+        ConfigFactory.parseMap(ImmutableMap.of("admin_oidc_enhanced_logout_enabled", "true"));
+
+    // Set up an admin account
+    AccountModel account = new AccountModel();
+    account.setGlobalAdmin(true);
+    // Explicitly set tokens Null to test behavior.
+    IdTokens idTokens = null;
+    account.setIdTokens(idTokens);
+    when(accountRepository.lookupAccount(accountId)).thenReturn(Optional.of(account));
+    Provider<AccountRepository> accountRepositoryProvider = () -> accountRepository;
+
+    OidcClientProviderParams params =
+        OidcClientProviderParams.create(civiformConfig, profileFactory, accountRepositoryProvider);
+    CiviformOidcLogoutActionBuilder builder =
+        new CiviformOidcLogoutActionBuilder(
+            oidcConfig, clientId, params, IdentityProviderType.ADMIN_IDENTITY_PROVIDER);
+
+    Optional<RedirectionAction> logoutAction =
+        builder.getLogoutAction(
+            new CallContext(getWebContext(), sessionStore), civiFormProfileData, targetUrl);
+
+    assertThat(logoutAction).isNotEmpty();
+    assertThat(logoutAction.get().getCode()).isEqualTo(302);
+
+    String location = ((FoundAction) logoutAction.get()).getLocation();
+    assertThat(location).isNotEmpty();
+    URI locationUri = new URI(location);
+    assertThat(locationUri.getHost()).isEqualTo(oidcHost);
+    assertThat(locationUri.getPort()).isEqualTo(oidcPort);
+    assertThat(locationUri.getPath()).isEqualTo("/session/end");
+
+    assertThat(queryParamValue(locationUri, "client_id")).hasValue(clientId);
+    assertThat(queryParamValue(locationUri, "post_logout_redirect_uri")).hasValue(targetUrl);
+
+    // No Serialized Tokens since token is null
+    Optional<String> serializedToken = queryParamValue(locationUri, "id_token_hint");
+    assertThat(serializedToken).isEmpty();
+  }
+
+  @Test
+  public void testBuilderLogsOutIfIdTokenNotFoundForAccount() throws URISyntaxException {
+    // Enable enhanced logout for admins.
+    Config civiformConfig =
+        ConfigFactory.parseMap(ImmutableMap.of("admin_oidc_enhanced_logout_enabled", "true"));
+
+    // Set up an admin account
+    AccountModel account = new AccountModel();
+    account.setGlobalAdmin(true);
+    // Assign token to session different from session being logged out to force an empty lookup.
+    IdTokens idTokens = new IdTokens(ImmutableMap.of("some_session_id", idToken));
+    account.setIdTokens(idTokens);
+    when(accountRepository.lookupAccount(accountId)).thenReturn(Optional.of(account));
+    Provider<AccountRepository> accountRepositoryProvider = () -> accountRepository;
+
+    OidcClientProviderParams params =
+        OidcClientProviderParams.create(civiformConfig, profileFactory, accountRepositoryProvider);
+    CiviformOidcLogoutActionBuilder builder =
+        new CiviformOidcLogoutActionBuilder(
+            oidcConfig, clientId, params, IdentityProviderType.ADMIN_IDENTITY_PROVIDER);
+
+    Optional<RedirectionAction> logoutAction =
+        builder.getLogoutAction(
+            new CallContext(getWebContext(), sessionStore), civiFormProfileData, targetUrl);
+
+    assertThat(logoutAction).isNotEmpty();
+    assertThat(logoutAction.get().getCode()).isEqualTo(302);
+
+    String location = ((FoundAction) logoutAction.get()).getLocation();
+    assertThat(location).isNotEmpty();
+    URI locationUri = new URI(location);
+    assertThat(locationUri.getHost()).isEqualTo(oidcHost);
+    assertThat(locationUri.getPort()).isEqualTo(oidcPort);
+    assertThat(locationUri.getPath()).isEqualTo("/session/end");
+
+    assertThat(queryParamValue(locationUri, "client_id")).hasValue(clientId);
+    assertThat(queryParamValue(locationUri, "post_logout_redirect_uri")).hasValue(targetUrl);
+
+    // No Serialized Tokens since token not found for account.
+    Optional<String> serializedToken = queryParamValue(locationUri, "id_token_hint");
+    assertThat(serializedToken).isEmpty();
+  }
+
+  @Test
+  public void testBuilderLogsOutIfAccountIsNotFound() throws URISyntaxException {
+    // Enable enhanced logout for admins.
+    Config civiformConfig =
+        ConfigFactory.parseMap(ImmutableMap.of("admin_oidc_enhanced_logout_enabled", "true"));
+
+    // Return empty account on lookup.
+    when(accountRepository.lookupAccount(accountId)).thenReturn(Optional.empty());
+    Provider<AccountRepository> accountRepositoryProvider = () -> accountRepository;
+
+    OidcClientProviderParams params =
+        OidcClientProviderParams.create(civiformConfig, profileFactory, accountRepositoryProvider);
+    CiviformOidcLogoutActionBuilder builder =
+        new CiviformOidcLogoutActionBuilder(
+            oidcConfig, clientId, params, IdentityProviderType.ADMIN_IDENTITY_PROVIDER);
+
+    Optional<RedirectionAction> logoutAction =
+        builder.getLogoutAction(
+            new CallContext(getWebContext(), sessionStore), civiFormProfileData, targetUrl);
+
+    assertThat(logoutAction).isNotEmpty();
+    assertThat(logoutAction.get().getCode()).isEqualTo(302);
+
+    String location = ((FoundAction) logoutAction.get()).getLocation();
+    assertThat(location).isNotEmpty();
+    URI locationUri = new URI(location);
+    assertThat(locationUri.getHost()).isEqualTo(oidcHost);
+    assertThat(locationUri.getPort()).isEqualTo(oidcPort);
+    assertThat(locationUri.getPath()).isEqualTo("/session/end");
+
+    assertThat(queryParamValue(locationUri, "client_id")).hasValue(clientId);
+    assertThat(queryParamValue(locationUri, "post_logout_redirect_uri")).hasValue(targetUrl);
+  }
 }

